test/builder/README.md

```diff
@@ -0,0 +1,120 @@
+# Builder package for tests
+
+This package holds `Builder` functions that can be used to create
+struct in tests with less noise.
+
+One of the most important characteristic of a unit test (and any type
+of test really) is **readability**. This means it should be easy to read
+but most importantly it should clearly show the intent of the
+test. The setup (and cleanup) of the tests should be as small as
+possible to avoid the noise. Those builders exists to help with that.
+
+There is two types of functions defined in that package :
+
+	* *Builders*: create and return a struct
+	* *Modifiers*: return a function
+	  that will operate on a given struct. They can be applied to other
+	  Modifiers or Builders.
+
+Most of the Builder (and Modifier) that accepts Modifiers defines a
+type (`TypeOp`) that can be satisfied by existing function in this
+package, from other package *or* inline. An example would be the
+following.
+
+```go
+	// Definition
+	type TaskOp func(*v1alpha1.Task)
+	func Task(name string, ops ...TaskOp) *v1alpha1.Task {
+		// […]
+	}
+	func TaskNamespace(namespace string) TaskOp {
+		return func(t *v1alpha1.Task) {
+			// […]
+		}
+	}
+	// Usage
+	t := Task("foo", TaskNamespace("bar"), func(t *v1alpha1.Task){
+		// Do something with the Task struct
+		// […]
+	})
+```
+
+The main reason to define the `Op` type, and using it in the methods
+signatures is to group Modifier function together. It makes it easier
+to see what is a Modifier (or Builder) and on what it operates.
+
+By convention, this package is import with the "tb" as alias. The
+examples make that assumption.
+
+# Example
+
+Let's look at a non-exhaustive example.
+
+```go
+package builder_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/knative/build-pipeline/test/builder"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func MyTest(t *testing.T) {
+	// You can declare re-usable modifiers
+	myStep := tb.Step("my-step", "myimage")
+	// … and use them in a Task definition
+	myTask := tb.Task("my-task", "namespace", tb.TaskSpec(
+		tb.Step("simple-step", "myotherimage", tb.Command("/mycmd")),
+		myStep,
+	))
+	// … and another one.
+	myOtherTask := tb.Task("my-other-task", "namespace", tb.TaskNamespace("foo"),
```

Comment:
Is `tb.TaskNamespace("foo")` left over by accident?

Reply:
indeed :man_facepalming: